src/flint/fmpq_mat.jl

```diff
@@ -210,6 +210,23 @@ function -(x::QQMatrix)
   return z
 end
 
+###############################################################################
+#
+#   Common denominator
+#
+###############################################################################
+
+# This function is really slow...
+function denominator(M::QQMatrix)
+  d = one(ZZ)
+  for i in 1:nrows(M)
+    for j in 1:ncols(M)
+      d = lcm!(d, d, denominator(M[i, j]))
```

Comment:
We should import `denominator!(z::ZZRingElem, y::QQFieldElem)` from Hecke. And then extend it to accept `QQFieldElemOrPtr`, and do something like
```suggestion
      d2 = denominator!(d2, mat_entry_ptr(M, i, j))
      d = lcm!(d, d, d2)
```
where `d2` is a second temporary. That's of course still not going to get speed records but at least it'll be a bit less wasteful.

Of course that's not a request for this PR, just musing about this code will browsing past it.